src/Plug.vala

```diff
@@ -367,10 +369,39 @@ namespace Power {
                 grid.attach (dim_label, 0, 0, 1, 1);
                 grid.attach (dim_switch, 1, 0, 1, 1);
             }
-
             return grid;
         }
 
+        private static bool lid_detect () {
+            var lid_path = File.new_for_path ("/proc/acpi/button/lid/");
+            var enumerator = lid_path.enumerate_children ("standard::*", 0);
+            try {
+                FileInfo lid_file;
+                if ((lid_file = enumerator.next_file ()) != null) {
+                    debug ("Detected lid switch");
+                    return true;
+                }
+            }catch (SpawnError err) {
+                    warning (err.message);
```

Comment:
use the right indentation and better use `critical` to report errors instead of `warning`